server/service/src/sync/translations/master_list.rs

```diff
@@ -59,6 +129,94 @@ impl SyncTranslation for MasterListTranslation {
     }
 }
 
+struct GenerateRequisitionProgram {
+    pub program_row: ProgramRow,
+    pub program_requisition_settings_row: ProgramRequisitionSettingsRow,
+    pub program_requisition_order_type_rows: Vec<ProgramRequisitionOrderTypeRow>,
+}
+
+fn generate_requisition_program(
+    connection: &StorageConnection,
+    master_list: &LegacyListMasterRow,
+) -> Result<Option<GenerateRequisitionProgram>, anyhow::Error> {
+    if master_list.is_program == false {
+        return Ok(None);
+    }
+
+    let program_settings = master_list.program_settings.clone().ok_or(anyhow::anyhow!(
+        "Program settings not found for program {}",
+        master_list.id
+    ))?;
+    let order_types = program_settings
+        .store_tags
+        .tags
+        .values()
+        .filter_map(|tag| tag.order_types.clone())
+        .flatten()
+        .collect::<Vec<LegacyOrderType>>();
+
+    let name_tag_and_period_schedule_ids = get_name_tag_and_period_schedule_id(
+        connection,
+        &program_settings.store_tags,
+        &master_list.id.clone(),
+    )?;
+
+    let program_row = ProgramRow {
+        id: master_list.id.clone(),
+        name: master_list.description.clone(),
+    };
+
+    let program_requisition_settings_row = ProgramRequisitionSettingsRow {
+        // Concatenate the program id and name tag id to create a unique id
+        // instead of using uuid since easier to test this way.
+        id: format!("{}{}", master_list.id, name_tag_and_period_schedule_ids.0),
+        name_tag_id: name_tag_and_period_schedule_ids.0,
+        program_id: master_list.id.clone(),
+        period_schedule_id: name_tag_and_period_schedule_ids.1,
+    };
+
+    let program_requisition_order_type_rows = if !order_types.is_empty() {
+        order_types
+            .iter()
+            .map(|order_type| ProgramRequisitionOrderTypeRow {
+                id: format!("{}{}", program_requisition_settings_row.id, order_type.name),
+                program_requisition_settings_id: program_requisition_settings_row.id.clone(),
+                name: order_type.name.to_string(),
+                threshold_mos: order_type.threshold_mos,
+                max_mos: order_type.max_mos,
+                max_order_per_period: order_type.max_order_per_period,
+            })
+            .collect::<Vec<ProgramRequisitionOrderTypeRow>>()
+    } else {
+        Vec::new()
+    };
+
+    Ok(Some(GenerateRequisitionProgram {
+        program_row,
+        program_requisition_settings_row,
+        program_requisition_order_type_rows,
+    }))
+}
+
+fn get_name_tag_and_period_schedule_id(
+    connection: &StorageConnection,
+    store_tag: &LegacyStoreTagAndProgramName,
+    id: &String,
+) -> Result<(String, String), anyhow::Error> {
+    let name_tag = NameTagRowRepository::new(connection)
+        .find_one_by_name(store_tag.tags.keys().next().unwrap())?
```

Comment:
What will `unwrap()` do if there are no store_tag.tags keys?

Reply:
I have this in my local...
```
{
    "storeTags": {
        "ProgramSchedule": {
            "periodScheduleName": "test_period_schedule",
            "orderTypes": []
        }
    },
    "elmisCode": ""
}
``

Reply:
Key is always the Program name, will add comment.

        "ProgramSchedule":  <- this